File: solutions/python/difference-of-squares/1/difference_of_squares.py
# In pure devotion to Shree DR.MDD

def square_of_sum(count):
    return sum(range(count + 1)) ** 2


def sum_of_squares(count):
    return sum(n * n for n in range(count + 1))


def difference_of_squares(count):
    return square_of_sum(count) - sum_of_squares(count)

# vim:ts=2:sw=2:expandtab

File: solutions/ocaml/difference-of-squares/1/difference_of_squares.ml
let square_of_sum n =
  let sum = (n * (n + 1)) / 2 in
  sum * sum
let sum_of_squares n =
  let sum = n * (n + 1) * (n * 2 + 1) in
  sum / 6
let difference_of_squares n = (square_of_sum n) - (sum_of_squares n)